messaging/WFLY-13793_disable_legacy_amq1_prefix.adoc

```diff
@@ -0,0 +1,126 @@
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1559[EAP7-1559]
+* https://issues.redhat.com/browse/EAP7-1244[EAP7-1244]
+
+Previous EAP7-1244 exposed the enable1Prefixes configuration on external pooled connection factories, but it does not allow to disable the use of legacy amq1 prefix for queue or topic defined in the model or using JakartaEE annotations.
+
+=== Dev Contacts
+
+* mailto:chaowan@redhat.com[Chao Wang]
+* mailto:ehugonne@redhat.com[Emmanuel Hugonnet]
+
+=== QE Contacts
+
+tborgato@redhat.com
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. Discuss with QE during the Kickoff state to decide this
+[X] Engineering
+
+[ ] QE
+
+* TBD
+
+=== Affected Projects or Components
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+* Add a `enable-amq1-prefix`  boolean attribute to the https://wildscribe.github.io/WildFly/18.0/subsystem/messaging-activemq/external-jms-queue/[/subsystem=messaging-activemq/external-jms-queue resource] to control the enablement of Artemis 1.x prefix for queue defined on a remote Artemis server. Default is `true` (meaning that name retro-compatibility is enabled). For instance an queue named `testQueue`, by setting this attribute to `false`, it effectively allows you to create an external queue without amq1 prefix `jms.queue`. The created queue will be `testQueue`. Otherwise, it will create an external queue with amq1 prefix `jms.queue`. The created queue will be `jms.queue.testQueue`.
+
+
+* Add a `enable-amq1-prefix`  boolean attribute to the https://wildscribe.github.io/WildFly/18.0/subsystem/messaging-activemq/external-jms-topic/[/subsystem=messaging-activemq/external-jms-topic resource] to control the enablement of Artemis 1.x prefix for topic defined on a remote Artemis server. Default is `true` (meaning that name retro-compatibility is enabled). For instance a topic named `testTopic`, by setting this attribute to `false`, it effectively allows you to create an external topic without amq1 prefix `jms.topic`. The created topic will be `testTopic`. Otherwise, it will create an external topic with amq1 prefix `jms.topic`. The created topic will be `jms.topic.testTopic`.
+
+* For a destination (queue or topic) created by annotation `@JMSDestinationDefinition` on a remote Artemis Server, add a `enable-amq1-prefix` boolean property to control the enablement of Artemis 1.x prefix. If the property is not defined, the default is `true` (meaning that name retro-compatibility is enabled).
+
+[source, java]
+----
+@JMSDestinationDefinition(
+    // explicitly mention a resourceAdapter corresponding to a pooled-connection-factory resource to the remote server
+    resourceAdapter = "remote-artemis",
+    name="java:global/env/myQueue2",
+    interfaceName="javax.jms.Queue",
+    destinationName="myQueue2",
+    properties = {
+        "enable-amq1-prefix=false"
+    }
+)
+----
+
+=== Hard Requirements
+
+N/A
```

Comment:
@soul2zimate RFE without any hard requirements doesn't make sense. Please fix it (maybe move text from the beginning of `Requirements` chapter here?)

Reply:
moved it from `Requirements` to `Hard Requirements`